From: David Milburn <dmilburn@redhat.com>
Date: Tue, 21 Jul 2009 07:48:21 -0500
Subject: [ahci] add SATA GEN3 related messages
Message-id: 20090721124821.GB6426@dhcp-210.hsv.redhat.com
O-Subject: [RHEL5.4 PATCH] ahci: add SATA GEN3 related messages
Bugzilla: 512086
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Jeff Garzik <jgarzik@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

The ahci driver doesn't report the correct link speed
for GEN3 hardware, AMD has verified a -158.el5 test kernel
built with this patch reports correct speed.

This resolves BZ 512086, please review and ACK.

Upstream patch from Shane Huang
commit 8522ee25f3a645577d41e71328cd4fcf8610dfeb

 drivers/ata/ahci.c        |    2 ++
 drivers/ata/libata-core.c |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/ata/ahci.c b/drivers/ata/ahci.c
index 0599d98..706ad81 100644
--- a/drivers/ata/ahci.c
+++ b/drivers/ata/ahci.c
@@ -2437,6 +2437,8 @@ static void ahci_print_info(struct ata_host *host)
 		speed_s = "1.5";
 	else if (speed == 2)
 		speed_s = "3";
+	else if (speed == 3)
+		speed_s = "6";
 	else
 		speed_s = "?";
 
diff --git a/drivers/ata/libata-core.c b/drivers/ata/libata-core.c
index 252e7dc..4ab96b5 100644
--- a/drivers/ata/libata-core.c
+++ b/drivers/ata/libata-core.c
@@ -859,6 +859,7 @@ static const char *sata_spd_string(unsigned int spd)
 	static const char * const spd_str[] = {
 		"1.5 Gbps",
 		"3.0 Gbps",
+		"6.0 Gbps",
 	};
 
 	if (spd == 0 || (spd - 1) >= ARRAY_SIZE(spd_str))
